### ISSUE TITLE
heatplot.ado does not verify whether colrspace is installed

### ISSUE THREAD

**Issue by michaelstepner** [benjann/heatplot#1]
According to the README:

> The palettes package and, in Stata 14.2 or newer, the colrspace package are required. To install these packages, type
> 
> . ssc install palettes, replace
> . ssc install colrspace, replace

The ado-file verifies that palettes is installed and provides a helpful error message if it is not. But it does not check whether colrspace is installed, which left me briefly confused as a new user:

```
capt which colorpalette
if _rc {
    di as err "the palettes package needs to be installed; type {stata ssc install palettes, replace}"
    exit 499
}
```

When `colrspace` was not installed, I received the following uninformative error message:
```
colrspace_library_palettes.sthlp not found
file not found
r(601);
```

**Comment by benjann** [comment]
Hi, thanks for reporting this. I now updated `heatplot` on github so that it will throw a better error message in this case. The change will also eventually make it into the version at the SSC archive.

Your comment actually pointed my to another problem. Installing `palettes` from SSC also installs file `lcolrspace.mlib` because in earlier days this was all that was needed from `colrspace` to make `palettes` (to be precise, the `colorpalette` command) work. However, at some point I revised `colrspace` and now additional files need to be installed (e.g. `colrspace_library_palettes.sthlp`). The issue now is that `colorpalette` looks for `lcolrspace.mlib` and throws an informative error message when not found; if found, however, `colorpalette` assumes that `colrspace` is installed and continues without error message, but then, later on, will bump into the uninformative error you reported.

In sent a note to the maintainer of the SSC archive to remove `lcolrspace.mlib`  from the installation files of the `palettes` package. This will help preventing confusions similar to the one you reported.

ben
